scipy/stats/_distn_infrastructure.py

```diff
@@ -2853,6 +2855,128 @@ def entropy(pk, qk=None, base=None, axis=0):
     return S
 
 
+def differential_entropy(
+    values: np.ndarray,
+    *,
+    window_length: Optional[int] = None,
+    base: Optional[float] = None,
+    axis: int = 0,
+) -> np.ndarray:
+    r"""Given a sample of a distribution, calculate the differential entropy.
+
+    This routine uses the Vasicek estimator of the differential entropy. Given
+    a sorted random sample :math:`X_1, \ldots X_n` this is defined as:
+
+    .. math::
+        \frac{1}{n}\sum_1^n \log\left[ \frac{n}{2m} (X_{i+m} - X_{i-m}) \right]
+
+    where :math:`m` is the window length parameter, :math:`X_{i} = X_1` if
+    :math:`i < 1` and :math:`X_{i} = X_n` if :math:`i > n`.
+
+    Parameters
+    ----------
+    values : sequence
+        Samples of the (continuous) distribution.
+    window_length : int, optional
+        Window length for computing Vasicek estimate. Must be an integer
+        between 1 and half of the sample size. If ``None`` (the default) it
+        uses the heuristic value
+
+        .. math::
+            \left \lfloor \sqrt{n} + 0.5 \right \rfloor
+
+        where :math:`n` is the sample size. This heuristic was originally
+        proposed in [2]_ and has become common in the literature.
+    base : float, optional
+        The logarithmic base to use, defaults to ``e`` (natural logarithm).
+    axis : int, optional
+        The axis along which the differential entropy is calculated.
+        Default is 0.
+
+    Returns
+    -------
+    entropy : float
+        The calculated differential entropy.
+
+    Notes
+    -----
+    This function will converge to the true differential entropy in the limit
+    when
+
+    .. math::
+        n \to \infty, \quad m \to \infty, \quad \frac{m}{n} \to 0
+
+    The optimal choice of ``window_length`` for a given sample size, depends on
+    the (unknown) distribution. In general, the smoother the density of the
+    distribution, the larger is such optimal value of ``window_length`` [1]_.
+
+    References
+    ----------
+    .. [1] Vasicek, O. (1976). A test for normality based on sample entropy.
+           Journal of the Royal Statistical Society:
+           Series B (Methodological), 38(1), 54-59.
+    .. [2] Crzcgorzewski, P., & Wirczorkowski, R. (1999). Entropy-based
+           goodness-of-fit test for exponentiality. Communications in
+           Statistics-Theory and Methods, 28(5), 1183-1202.
+
+    Examples
+    --------
+
+    >>> from scipy.stats import differential_entropy, norm
+
+    Entropy of a standard normal distribution:
+
+    >>> random_state = np.random.RandomState(0)
+    >>> values = random_state.standard_normal(100)
+    >>> differential_entropy(values)
+    1.342551187000946
+
+    Compare with the true entropy:
+
+    >>> float(norm.entropy())
+    1.4189385332046727
+
+    """
+    values = asarray(values)
+    values = np.moveaxis(values, axis, 0)
+
+    if window_length is None:
+        window_length = math.floor(math.sqrt(len(values)) + 0.5)
```

Comment:
```suggestion
        window_length = np.int64(np.floor(np.sqrt(len(values)) + 0.5))
```
Since NumPy is already imported?

Reply:
What would be the advantage of doing that?

Reply:
None to code execution. Maybe controversial comments will attract another reviewer : )

Reply:
So... should I change this then? Because the conversation is marked as solved. And also changing the type of `window_length` in the middle of the code from `int` to `np.int64` makes Mypy cry.

Reply:
If it's a Python scalar that will only ever be used as a Python scalar, using the `math` module is fine. Especially if you otherwise need to apply another band-aid for mypy.